.travis.yml

```diff
@@ -1,31 +1,25 @@
 language: ruby
 rvm: 2.0.0
-sudo: required
-os: linux
 
-matrix:
-  include:
-    - os: osx
-      env: OSX=10.11
-      osx_image: xcode7.3
-      rvm: system
-  allow_failures:
-    - os: linux
+env:
+  global:
+    - PATH="~/.linuxbrew/bin:~/.linuxbrew/sbin:$PATH"
+    - HOMEBREW_DEVELOPER=1
+    - LINUXBREW=$HOME/.linuxbrew
 
 before_install:
   - if [ -f ".git/shallow" ]; then travis_retry git fetch --unshallow; fi
-  - if [ "${TRAVIS_OS_NAME}" = "linux" ]; then
-      git clone https://github.com/Homebrew/linuxbrew.git ~/.linuxbrew;
-      PATH="~/.linuxbrew/bin:~/.linuxbrew/sbin:$PATH";
-    fi
-  - sudo chown -R $USER $(brew --repo);
-  - brew update || brew update;
-  - if [ "${TRAVIS_OS_NAME}" = "linux" ]; then brew tap linuxbrew/xorg; fi
-  - mkdir -p $(brew --repo)/Library/Taps/homebrew;
-  - ln -s $PWD $(brew --repo)/Library/Taps/homebrew/homebrew-emacs;
-  - cd $(brew --repo)/Library/Taps/homebrew/homebrew-emacs;
-  - export TRAVIS_BUILD_DIR="$(brew --repo)/Library/Taps/homebrew/homebrew-emacs";
-  - export HOMEBREW_DEVELOPER="1"
+  - HOME=/home/linuxbrew
+  - sudo mkdir $HOME
+  - sudo chown "$USER:" $HOME
+  - mkdir $LINUXBREW
+  - curl -L https://github.com/Linuxbrew/brew/tarball/master | tar xz -m --strip 1 -C $LINUXBREW
```

Comment:
@sjackman, a lot of these commands will probably need to be executed in the Travis file for every single tap. Counting core, science, dupes, xorg, and emacs, that's 5, and there might be other taps that I'm not aware of.

What do you think about extracting some of these commands into a gist to reduce the amount of syncronization we need to do between repos?